source/s2n/s2n_tls_channel_handler.c

```diff
@@ -67,6 +117,21 @@ void aws_tls_clean_up_thread_local_state(void) {
 
 void aws_tls_clean_up_static_state(void) {
     s2n_cleanup();
+
+#if OPENSSL_VERSION_LESS_1_1
+    if (CRYPTO_get_locking_callback() == s_locking_fn) {
+        CRYPTO_set_locking_callback(NULL);
```

Comment:
should we statically store whether or not we were the ones to install the locking callback?

as is, we don't set it if it was already there, but we'll remove it either way

Reply:
doesn't this branch check exactly that? The only way we remove it is if the locking callback is the one we would have installed.